Code/Tools/FBuild/FBuildCore/BFF/Functions/FunctionObjectList.cpp

```diff
@@ -470,8 +470,15 @@ bool FunctionObjectList::GetInputs( const BFFIterator & iter, Dependencies & inp
 		inputs.Append( dirNodes );
 	}
 
+	const char * rootDirectory = nullptr;
+	const BFFVariable * compilerInputFilesRoot = nullptr;
+	if ( GetString( iter, compilerInputFilesRoot, ".CompilerInputFilesRoot", true ) )
+	{
```

Comment:
We need to return false here if GetString returns false as that indicates a parsing error has occurred.